toonz/sources/toonzlib/tstageobjectspline.cpp

```diff
@@ -255,30 +263,31 @@ void TStageObjectSpline::loadData(TIStream &is) {
     }
   }
 
-    // try loading the old type
-    if (points.size() == 0) { // Needed for T2D compatibility 
-        while (!is.eos()) {
-            TThickPoint p;
-            is >> p.x >> p.y >> p.thick;
-            points.push_back(p);
-        }
-        m_interpolationStroke.clear();
-        m_interpolationStroke.push_back(TPointD(-40, 0));
-        m_interpolationStroke.push_back(TPointD(-20, 0));
-        m_interpolationStroke.push_back(TPointD(-20, 0));
-        m_interpolationStroke.push_back(TPointD(0, 0));
-        m_interpolationStroke.push_back(TPointD(65, 65));
-
-        m_interpolationStroke.push_back(TPointD(935, 935));
-        m_interpolationStroke.push_back(TPointD(1000, 1000));
-        m_interpolationStroke.push_back(TPointD(1020, 1000));
-        m_interpolationStroke.push_back(TPointD(1020, 1000));
-        m_interpolationStroke.push_back(TPointD(1040, 1000));
-        m_active = false;
-        m_steps = 10;
-        m_color = 0;
-        m_width = 1;
+  // try loading the old type
+  if (m_externalDataFound &&
+      points.size() == 0) {  // Needed for T2D compatibility
```

Comment:
I don't think this block is needed.
The original commit in T2D (https://github.com/tahoma2d/tahoma2d/pull/420/), this condition was added replacing [the block for handling old version of Toonz scene files](https://github.com/tahoma2d/tahoma2d/pull/420/files#diff-455d1a948001440c636d62a3294f844e47eb8e5258c07936bc4a0bc5a9908c80L178-L184) .
In OT the conventional version check is still there.

Reply:
Removed.   I was skeptical about including it myself. Not even sure why it was added to T2D as it seems an unlikely scenario.

I squashed my commits and forced pushed because I realized I forgot to add a co-authored-by Jeremy since I took these changes from his Motion Path PR.